0022-hashed-timelock-agreements/0022-hashed-timelock-agreements.md

```diff
@@ -54,21 +54,36 @@ For more details on HTLAs without ledger support see the sections below on [Simp
 
 A single Interledger payment can cross multiple different types of HTLAs. The Interledger Protocol ensures that the security of each HTLA is independent of the other HTLA types used in the path.
 
-For example, in this rendition of the classic Alice to Bob payment, Alice has an account on a blockchain that implements HTLCs and Bob has an account with a bank that does not implement HTLCs:
+This is a critical point for understanding HTLAs and Interledger, so we will use an example flow to illustrate it.
+
+### Example Payment Flow
+
+In this rendition of the classic Alice to Bob payment, Alice has an account on a blockchain that implements HTLCs and Bob has an account with a bank that does not implement HTLCs:
 
 ```
-        On-Ledger HTLC               Payment Channel               Trustline
-Alice --(Blockchain A)--> Connector --(Blockchain B)--> Connector --(Bank C)--> Bob
+        On-Ledger HTLC                 Payment Channel                  Trustline
+Alice --(Blockchain A)--> Connector 1 --(Blockchain B)--> Connector 2 --(Bank C)--> Bob
 ```
 
-Each `--(...)-->` represents a different type of HTLA, but all of them use the same hashlock. Each connector forwards the payment, preparing funds with their peer, and passes back the preimage once it is produced by Bob.
+Each `--(...)-->` represents a different type of HTLA, but all of them use the same hashlock.
+
+The Interledger protocol flow (detailed below) ensures that each participant only needs to care about the HTLAs in which they are directly involved. Failure in a different part of the path -- or the selection of a seemingly unwise HTLA type -- does not affect any other agreements.
 
-This setup ensures that Alice only needs to care about whether her ledger executes the HTLC faithfully. If so, her funds will only be transferred once Bob has been paid.
+#### Interledger Protocol payment flow:
+
+1. `Alice` and `Bob` agree on the hashlock `H`. The preimage `P` is only known to `Bob`.
+2. `Alice` prepares the transfer to `Connector 1` by creating and funding an HTLC on `Blockchain A` with hashlock `H`.
+3. `Connector 1` prepares a transfer to `Connector 2` via their shared payment channel, also using hashlock `H`.
+4. `Connector 2` prepares a transfer to `Bob` on their shared trustline using Hashlock `H`.
+5. If `Bob` produces the preimage `P` before the transfer timeout, `Connector 2` will "pay" `Bob` by increasing his balance on their trustline.
```

Comment:
This should go through the whole chain of the success case, and then have a separate list of steps describing how the failure case doesn't affect anyone except the failing link